### ISSUE TITLE
Adding self-recurrence (time-travel) trick to schedulers

### ISSUE THREAD

**Issue by anvilarth** [huggingface/diffusers#4575]
**Is your feature request related to a problem? Please describe.**
In image restoration problems like (image-inpainting, super-resolution, etc) it's common practice to use this trick to improve quality of generated images. 

**Describe the solution you'd like**
Adding parameters in schedulers which allow to apply time travel trick

**Comment by sayakpaul** [comment]
Could you provide the following?

* Reference to the trick
* Some comparative results with and without this trick 

We'd be happy to review any PRs for this feature if there's mutual consensus. Ccing @yiyixuxu and @williamberman. WDYT?

**Comment by williamberman** [comment]
Yeah I'm not familiar either would appreciate a reference. Fwiw we generally have to be careful about adding new features to the codebase even if they're cool just to try to manage complexity. So can't give an ahead of time yes or no before hearing more :)

**Comment by anvilarth** [comment]
Hmm, looks like it has been already implemented in here:
https://huggingface.co/docs/diffusers/v0.19.3/en/api/schedulers/repaint#diffusers.RePaintScheduler

The difference in naming confuses a lot :)

Thx, for help!
